File: CleanJavaRx/app/src/releaseUnitTests/java/com/vascome/fogtail/ui/base/other/NoOpViewModifierTest.java
package com.vascome.fogtail.screens.base.other;

import android.view.View;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verifyZeroInteractions;

public class NoOpViewModifierTest {

    @Test
    public void modify_shouldReturnPassedViewAndNotInteractWithIt() {
        View view = mock(View.class);

        assertThat(new NoOpViewModifier().modify(view)).isSameAs(view);
        verifyZeroInteractions(view);
    }
}

File: MVVMKotlin/app/src/releaseUnitTests/java/com/vascome/fogtail/ui/base/other/NoOpViewModifierTest.kt
package com.vascome.fogtail.presentation.base.other

import android.view.View

import org.junit.Test

import org.assertj.core.api.Assertions.assertThat
import org.mockito.Mockito.mock
import org.mockito.Mockito.verifyZeroInteractions

class NoOpViewModifierTest {

    @Test
    fun modify_shouldReturnPassedViewAndNotInteractWithIt() {
        val view = mock(View::class.java)

        assertThat(NoOpViewModifier().modify(view)).isSameAs(view)
        verifyZeroInteractions(view)
    }
}